crates/forge/tests/fixtures/zk/NonceMismatch.t.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.18;
+
+import "forge-std/Test.sol";
+import {Greeter} from "../src/Greeter.sol";
+
+contract NonceMismatchTest is Test {
+    function setUp() public {
+        // Deploy contracts in setup to increment nonce
+        new Greeter();
+        new Greeter();
+        new Greeter();
+        new Greeter();
+    }
+
+    function testTxOriginNonceDoesNotUpdate() public {
```

Comment:
echoing @elfedy, perhaps we can clarify with a dedicated test that setup doesn't increment the nonce either

Reply:
Something like: `testTxOriginNonceDoesNotUpdateInSetup() public { assertEq(vm.getNonce(address(tx.origin)), 2); }`

Reply:
or like it's done later in the script, storing the original nonce and checking it doesn't increase during and after setup

Reply:
I'd also clarify that we are still increasing the test's deployment nonce, hence why we can deploy multiple times the same thing